includes/Utilities/Plugin_Request_Utility.php

```diff
@@ -159,4 +160,20 @@ public static function get_directories_to_ignore() {
 
 		return $directories_to_ignore;
 	}
+
+	/**
+	 * Checks if single file plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @return bool true if the single file plugin, otherwise false.
+	 */
+	public static function is_single_file_plugin( Check_Result $result ) {
+		if ( $result->plugin()->path() === $result->plugin()->location() ) {
+			return false;
+		}
+
+		return true;
+	}
```

Comment:
I'm not sure this class is the most suitable place for this method. Why not add this to the `Plugin_Context` class (and thus also make it available in `Check_Context`? This method only concerns the plugin specifically, so it should be part of the relevant class. Then you can simplify the logic to:
```php
return $this->path() !== $this->location();
```

Reply:
Thanks for feedback. Moved function in https://github.com/WordPress/plugin-check/pull/319/commits/7eea1044a27d047d56c1679d8b55af104a1847ff